include/zephyr/device.h

```diff
@@ -297,11 +298,22 @@ typedef int16_t device_handle_t;
  * @param compat lowercase-and-underscores devicetree compatible
  * @return a pointer to a device, or NULL
  */
-#define DEVICE_DT_GET_ANY(compat)					    \
+#define DEVICE_DT_GET_ONE_OR_NULL(compat)				    \
 	COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(compat),			    \
 		    (DEVICE_DT_GET(DT_COMPAT_GET_ANY_STATUS_OKAY(compat))), \
 		    (NULL))
 
+/**
+ * @brief Get a <tt>const struct device*</tt> from a devicetree compatible
+ *
+ * @param compat lowercase-and-underscores devicetree compatible
+ * @return a pointer to a device, or NULL
+ *
+ * @deprecated Use DEVICE_DT_GET_ONE_OR_NULL() instead.
+ */
+#define DEVICE_DT_GET_ANY(compat) __DEPRECATED_MACRO \
```

Comment:
As explained above I think deprecating it is doing more harm than good. Consistency is fine, but let's also consider the churn for the user. This doesn't cost anything at all to maintain.

Reply:
I think inconsistency also comes with a cost, and this is not being considered here. By looking at samples, we now have a mix of `ANY`/`ONE`, and it basically depends on what the user thought it was better at that moment, or what reviewer suggested. Reason for one or the other in most cases? None. If, as a new user, I write a sample, I'll probably end up picking the one from the sample I copied&pasted. It reminds me of DEVICE_DT_GET/device_get_binding. We finally brought consistency in-tree when `label` was deprecated. 

>> any person who comes in the store today is eligible for the discount

> [merriam-webster.com/dictionary/any](https://www.merriam-webster.com/dictionary/any)
>
> If zero people walk into the store, zero people will receive a discount. So 'any foo' can be used in contexts where no foo exists. This is why I chose that word when naming this API. So I don't think it's really fair to say it's "wrong". But I do agree it is inconsistent with the node_id versions so I have no objections to remedying that.

I'm not the language expert here, but when I read "any", I read "give me any of the devices, no matter which one", but I also understand the same for the "one" version. If API was a function and not a macro, both could be documented to return `NULL`. If we actually had constexpr functions, we would likely have a single function. So, it sounds like we have `ANY`/ONE` because of C language limitations. And if that's true, I'd make things clear using `TRY` or `OR_NONE`.